scripts/pylib/twister/twisterlib/size_calc.py

```diff
@@ -224,3 +264,169 @@ def _calculate_sizes(self):
             self.sections.append({"name": name, "load_addr": load_addr,
                                   "size": size, "virt_addr": virt_addr,
                                   "type": stype, "recognized": recognized})
+
+    def _get_footprint_from_elf_file(self) -> None:
+        self._check_elf_file()
+        self._check_is_xip()
+        self._calculate_from_elf_sections()
+
+    def _get_buildlog_file_content(self) -> typing.List[str]:
+        """Get content of the build.log file.
+
+        @return Content of the build.log file (list[str])
+        """
+        if os.path.exists(path=self.filename) and self.filename.endswith("build.log"):
+            with open(file=self.filename, mode='r') as file:
+                file_content = file.readlines()
+        else:
+            logger.error("Incorrect build.log file to analyze footprints.")
+            file_content = []
+        return file_content
+
+    def _find_idx_of_last_pattern_occurrence(self, file_content: typing.List[str], pattern: str) -> int:
+        """Find the index from which the information about the memory footprint is read.
+
+        @param file_content (list[str]) Content of build.log.
+        @param pattern (str) String with start phrase of memory footprint section.
+        @return: Index with information about the memory footprint (int)
+        """
+        result = 0
+        if not pattern:
+            pattern = self.PATTERN_SEARCHED_LINE
+
+        # Check the file in reverse order.
+        for idx, line in enumerate(reversed(file_content)):
+            # Condition is fulfilled if the pattern matches with the start of the line.
+            if re.match(pattern=pattern, string=line):
+                result = idx
+                break
+        # If the file does not contain information about memory footprint, the warning is raised.
+        if result == 0:
+            logger.error(msg="Information about memory footprint for this test configuration is not found.")
```

Comment:
What happen next, when this patter will not be found? As I understand `_find_idx_of_last_pattern_occurrence` will return index number as 0 what is unfortunately valid number for indexing and as a result program will try to continue calculations basing on this index. Maybe by default `result` variable should have value `None` what explicitly points that index was not found. Then in `_create_data_table` method should be applied extra verification: if found index is `None`, then calculation should be stopped.

Reply:
Done